### ISSUE TITLE
How to use and future plans?

### ISSUE THREAD

**Issue by Romkabouter** [recom3/api-hud-goggle#1]
Hi there! 

I still have an Oakley Airwave 1.5, I came across your site and was wondering what I can do with this API?
A long tome ago I created an app for the Snow2, but then Recon died and battery life sucked.
Are you having plans to revive the appstore as well?

So what do I get besides the app when I throw you some bitcoin? 

Thanks for the effort anyway!

**Comment by recom3** [comment]
Many thanks for writing!

We will try to clarify questions:

What besides the app? Now we have also the Uplink Desktop App (Windows). This Desktop App is able to synchronize trips to backend, so you have them available in Phone App.
Further, the Uplink App is able to install new apps in the goggles. At this moment we have only one app to install into the goggles: Un-blocker App. This Un-blocker allows users with factory blocked Airwaves (maybe also Jet’s, but we have not tried) to avoid the cancel update screen, every time they power up.

So… plans to revive the appstore? Yes! In fact we have only to include it in the private area of recom3.com
Technically, we can install 3rd party apps in the goggles at this moment. You can submit us your app and we will make it available for everyone.

The sketch API we have supplied is essential to support the Phone App’s. At this moment is mainly demonstrative, but will allow others, soon we hope, to develop Android/iOS on their own. With this API we are bale, for example, to show your friends position. We have already tested this functionality with two Airwave's!

So our plans is to revive the full ecosystem, but as you can see is a huge effort indeed! Everything was dead and basically is doing everything from zero!

Just to help you with the battery. It seems others were able to attach a generic battery:
[https://www.ifixit.com/Respuestas/Ver/365189/How+do+I+change+the+battery+in+a+recon+HUD+ski+goggle](https://www.ifixit.com/Respuestas/Ver/365189/How+do+I+change+the+battery+in+a+recon+HUD+ski+goggle)

Feel free to ask any further questions.

Regards!

**Comment by Romkabouter** [comment]
Good news and thanks for the work, I am willing to pay for the services however I am very reluctant about bitcoins.

> You can submit us your app and we will make it available for everyone.

Here is the github, I will see if I have the apk somewhere on my drive. It is a while ago, so building it might be an issue. I will surely put some effort into it.

**Comment by recom3** [comment]
Yup! We understand the reticence about BTC. In fact the costs are not covered at all (maybe 1%...), so we have put the obtained funds to yield generating. We thought it can be nice to have this funds for future, as something common to all.
We will put a PayPal donate button or similar. Maybe crowdfunding can work also.

You project is quite good! We can build it. No problem. Many thanks.
Just one thing, if you have the Recon SDK as jar or zip, please attach it to the project, because it seems newer as the version we have.

**Comment by Romkabouter** [comment]
That is a long while ago, but I think I was using this
https://github.com/Romkabouter/Snow2Liftie/tree/master/sdk-master

Is that what you are looking for? Dunno if it is the same version, but I do remember getting a new version indeed.

**Comment by Romkabouter** [comment]
I also created another app, controlling the Snow2 with the Myo Armband

https://www.youtube.com/watch?v=PtlgW2gx_dg&ab_channel=RomKabouter

**Comment by Romkabouter** [comment]
I have shared some jars here:
https://romkabouter.stackstorage.com/s/lmwQDS1MpVZqp9eR

Currently I do not now what is what, but I am still using the Snow2, it is build in my Oakley Airwave which I find an very good goggle.
I stopped using the HUD because regular app were better and battery life was crap. I will try if I can gear up the goggle with adb and see if I can get some things running.

**Comment by Romkabouter** [comment]
Also, the SDK is here as well I think
https://github.com/relet/Recon-HUD-Paragliding

**Comment by recom3** [comment]
Ey! Many thx!
We are taking a look to all the stuff.

**Comment by Romkabouter** [comment]
> We will put a PayPal donate button or similar. Maybe crowdfunding can work also.

It might be a good idea to let people have a free trail period, just to let them know you guys are serious with this  :)
Luckily I have exported all my tracks, but the Snow2 will not yet boot. Most probably because the battery is below the 2v threshold for charging, but we'll see. 

Keep up the good work, I am a software engineer so maybe I am able to put some time in this. Interested in the API for sure :)

**Comment by recom3** [comment]
Hello,
Free trial for sure, we have included in Q&A. At least 1 month. After that if you are not happy, we will send back the funds.
There is a formular in Q&A, under "What happens when my purchase is failing or I have a doubt?". Here you can say you are not happy and paste the transfer hash for return.

In fact, as I have said, we will put all the funds to yield farming in an exchange and make it public. We are fully committed with this project.

We are thinking about having a battery replacement section in the web. I have read again through  [ifixit](https://www.ifixit.com/Respuestas/Ver/365189/How+do+I+change+the+battery+in+a+recon+HUD+ski+goggle#answer726739) and seems the most promising model is the Li-Pol 103040 (seems easy to get one in the Web).

Let us now please if it works for you! 

Yeah! Sure we will need help in the future, as there are many things to do. We'll keep you posted!

**Comment by Romkabouter** [comment]
> Free trial for sure, we have included in Q&A.

Yeah, I have read that after my comment indeed. But still that means payment first, I think that will keep people from registering but that is just my opinion :)
I you can put some donation button on it would be great, because I am will to pay but I will not create a wallet for this at the moment. The process is overly complicated this way.

> Let us now please if it works for you!

I have gently cracked open the battery box, so progress there :)

**Comment by recom3** [comment]
Hello,
Most probably we will have to rebuild the recon sdk from scratch., at least the Bluetooth communications part.
That is going to be a nice piece of work...

I have created here a separate issue for the battery topics.

**Comment by Romkabouter** [comment]
Hi there, I have succesfully replaced the battery and would like to have access to the app and the reconlink.
Is there any other way I could pay at the moment?

**Comment by Romkabouter** [comment]
> Now we have also the Uplink Desktop App (Windows).

I can confirm this app works :)
The phone app also worked and I have paired the goggle to my phone. 

Should I already be able to see the runs in the phone app or is this not ready yet? I currently see 1 example run, might be because I do not have any runs anymore in the goggle

**Comment by Romkabouter** [comment]
I have searched and found my old recons apk files, I can upload them if you want. They can be decompiled but maybe you already have them

<img width="1002" alt="image" src="https://user-images.githubusercontent.com/4774478/215162851-007a0858-66b9-4bd9-9863-756aefa6e24c.png">

**Comment by recom3** [comment]
Hello,
Yes, please. Upload them to a shared folder, if you are so kind, so we can have a look.
About the trips, yes they should work in the app. First you have to upload them with the Uplink (you are right, when there are no trips it shows a sample).
In fact we are know working hard to allow to view the trips in a map in the private area of the website.
Regards.

**Comment by Romkabouter** [comment]
ok nice, I will upload them to the shared folder I mentioned earlier, along with the decompiled apps and other old stuff you might find usable (or not)

**Comment by recom3** [comment]
> ok nice, I will upload them to the shared folder I mentioned earlier, along with the decompiled apps and other old stuff you might find usable (or not)

Thx a lot!

**Comment by Romkabouter** [comment]
> cause this recon sdk is used mainly to communicate the Goggles to the Internet using Bluetooth and the Phone.

Yeah, and it also holds some of the UI stuff you can use to create apps. Well, that is what I can remember anyway.

**Comment by Romkabouter** [comment]
Nice, I will try and upload a trip!

**Comment by Romkabouter** [comment]
Firsts try, I have downloaded the sync app and synced with the site.
When I open the web beta, I can see my trip in de console, but a blank screen

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/4774478/216172127-eb309309-86b0-4ba1-8a71-72b77077b418.png">

Then when I refresh I get an API error

<img width="873" alt="image" src="https://user-images.githubusercontent.com/4774478/216172378-63533683-8c58-4884-85d4-3df653ddb976.png">

**Comment by recom3** [comment]
Fixed. Sorry for that!
Now Google limits everything...

**Comment by Romkabouter** [comment]
Cool, I'll try again after my holiday.
Maybe it is better to shift towards leaflet https://leafletjs.com/

Also, if you are up for it, it might be a good idea to OS the web part as well, so other devs could join. Depending on your plans obviously :)

**Comment by recom3** [comment]
Good day,
That is an interested question.
This Unblocker we have only tested with the Snow2 and is able to bypass the activation screen.
As we don't have any Jet's blocked, we have not tested it on Jet's...

Ok first think we are going to do a small change and be back. When we are ready we will write here. Thx.

Possibly we will upload the source code, yes.

Regards.

**Comment by recom3** [comment]
Another thing that could help, please activate the debug option in the glasses and reconnect them with Uplink Desktop.
Thx.

**Comment by Romkabouter** [comment]
> Another thing that could help, please activate the debug option in the glasses and reconnect them with Uplink Desktop. Thx.

You mean the Jet? That is not possible. I cannot do anything since I am not able to get passed the activation screen. I tried some reverse engineering as well with the lxl files but at the moment without result. I do not know what is needed to trigger activation.

**Comment by recom3** [comment]
> > Another thing that could help, please activate the debug option in the glasses and reconnect them with Uplink Desktop. Thx.
> 
> You mean the Jet? That is not possible. I cannot do anything since I am not able to get passed the activation screen. I tried some reverse engineering as well with the lxl files but at the moment without result. I do not know what is needed to trigger activation.

Yes you are right! Sorry.
So the Unblocker App should be installed by the Uplink Desktop, right?
Do you see anything in the screen of the glasses, apart from the Activate screen?
Describe the sequence please: 1st Warning screen, then...
Thx in advance.

**Comment by Romkabouter** [comment]
> So the Unblocker App should be installed by the Uplink Desktop, right?
> Do you see anything in the screen of the glasses, apart from the Activate screen?

Yes, the ublocker is indeed install.

1. Boot the Jet and connect Jet to Win11. -> Jet shows activation screen
2. Start Uplink
3. In the Jet a message appears, that new App are available (Recon3Hub) and if I want to install it.
4. On succesfull install, I can click done or open
5. If I open it, it says the App is running and I can press back
6. back -> return to activation screen
7. reboot Jet -> activation screen

**Comment by recom3** [comment]
Ok not so bad. At least the new App message appear, that is good.
After installing the App, do not press the back button, as it should close alone itself.
Before Activation screen does the glasses show the Warning Safety screen? The Warning Safety screen should be the first, as it is in the Snow2.

**Comment by Romkabouter** [comment]
> do not press the back button, as it should close alone itself.

Yes, it does.

> Before Activation screen does the glasses show the Warning Safety screen? The Warning Safety screen should be the first, as it is in the Snow2.

Nope, Activation screen right at boot. No Warning Safety. Might be due to software versions? I have bought it with a huge discount, to there might be an older version or something.

**Comment by recom3** [comment]
Extrange. Should be a Warning screen.
Which are the exact words from the Activation screen?

**Comment by recom3** [comment]
The flow in Not activated Snow2 is:
1. Screen WARNING: Always be aware of your surroundings....--> click OK
2. Screen Activate your device --> do nothing, wait
3. Unblocker por ups and closes automatically
4. GO
It seems that Jet is harder and has the Activation screen on top of everything

**Comment by Romkabouter** [comment]
I got some more info here:

https://forum.xda-developers.com/t/recon-jet-basic-info.3086703/

My Jet shows this (no back button and also no reaction os pressing back)
![image](https://user-images.githubusercontent.com/4774478/221192230-00a36d31-e962-47e3-88a6-ddcf1ab3730e.png)

So different from this:
![image](https://user-images.githubusercontent.com/4774478/221192320-187f6c85-b8de-47a0-9117-cb1ea441bb21.png)

Like I said, this might be due to other versions of the Recon software. There was a ReconsOS4, but also a version before that. I have extracted all my files from the Snow2 and will compare them with the Jet.

**Comment by recom3** [comment]
Maybe let's open an issue for Jet activation problem
We will check, but the good point if the unblock app is installing. Maybe with a little tuning...

**Comment by recom3** [comment]
Added map tile support for Recon Jet Glasses and Snow 4.5 (i.e. produce tiles from OSM open street exported maps).
Details in the other thread: https://github.com/recom3/api-hud-goggle/issues/3#issuecomment-1519187451

**Comment by recom3** [comment]
Hello,
After lot of effort the new Android App Ice3 is available to download.
This App allows web communications for the goggles.
As a sample it has been forked:
[Snow2Liftie](https://github.com/recom3/Snow2Liftie/blob/master/README.md )
Snow2Liftie -> credit to @Romkabouter (many thx!)
No recon sdk needed for communications any more.
Thx to everyone supporting us.

**Comment by recom3** [comment]
Source code of new Android app published here (still some draft parts, we apologize)
[https://github.com/recom3/Ice3](https://github.com/recom3/Ice3)

Now in recom3 site new 0.1 subscription for 3 months, that allows unlocking of Recon Jet!

**Comment by recom3** [comment]
September Update  
The release of new app Ice3 allows us to progress further and integrate existing applications in the device.  
We are opening a new market place with following free apps (many thx to the original developers):  
Snow2Liftie->shows lift status near in your favorite resort  
MyWeather2.0->shows weather conditions  
Unblocker App->existing app to unblock Jet Glasses

**Comment by recom3** [comment]
October Update  

We continue with the publication of the api source code and plan to be ready for this season.
Start to publish **Uplink Desktop**.  

All the code will be open sourced as promised.  
The fees are only to support the servers if you want to sync trips, buddies, apps, etc.  
We will plan to keep the lifetime support fee only this season (we will give service to the existing ones ofc but will not be available for new ones).

**Comment by recom3** [comment]
January Update

We have been working with Maps version 1 (first version of Recon Snow2). It is a vectorial format not used afterwards, but for the shake of completeness we wanted to fully implement it.
In this format as described in Maps project, the piste plan for a resort has to be implemented fully in QGIS and then exported to the format supported for the Goggles.

Progressing with **Uplink Desktop**, that is not fully ready to be open sourced, but vey close. In the meantime we want to implement the functionality to upgrade the Snow2 app to the last published version (packages for the upgrade been tested now).

Other possibility in the long run could be a hardware upgrade using newer processors  W4100. For that we plan to acquire some kits: TurboX SDW4100 Development Kit.

**Comment by Romkabouter** [comment]
Great! I am heading to the snow in a couple of weeks, so already recharged the goggle. Will try the new stuff asap!

Hardware upgrades sound really awesome, but do you mean refit it in the Snow2 or something?

**Comment by recom3** [comment]
> Great! I am heading to the snow in a couple of weeks, so already recharged the goggle. Will try the new stuff asap!
> 
> Hardware upgrades sound really awesome, but do you mean refit it in the Snow2 or something?

Well there are possibilities of improvement using the same form factor:

1. For processor box (Snow2) a better hardware with faster processor can be fit. Possibly Snapdragon W4100
2. Improve the current display with same, but with more resolution: Kopin KD030308KLV-K, resolución 854x480 (original was 480x240).

What do you think? At least keeping the form factor should be useful for the moment.

**Comment by Romkabouter** [comment]
I think it would be cool, but also I think that it is a very very small part of the people who own this device and actually still use it, are able to change the hardware.
And yes, form factor is very important, otherwise it will not fit I guess

**Comment by Romkabouter** [comment]
I still have a version 3.2.1 of the software on anothet goggle I got. I cannot seem to get it on the version OS4 unfortunatly

**Comment by recom3** [comment]
Mm still pending OS version thats true

**Comment by recom3** [comment]
> I think it would be cool, but also I think that it is a very very small part of the people who own this device and actually still use it, are able to change the hardware. And yes, form factor is very important, otherwise it will not fit I guess

Sure not a lot of people, but we think about the manufacturers existing experience like Oakley, Zeal, Smith IO to integrate the hardware.
In any case, would be nice to split not only battery, but too the visor unit itself from the CPU, to have even more flexibility.

The investment is high and we would have to start with a module like: TurboX SDW4100 Development Kit (that is alone 1000$).

**Comment by recom3** [comment]
> > I think it would be cool, but also I think that it is a very very small part of the people who own this device and actually still use it, are able to change the hardware. And yes, form factor is very important, otherwise it will not fit I guess
> 
> Sure not a lot of people, but we think about the manufacturers existing experience like Oakley, Zeal, Smith IO to integrate the hardware. In any case, would be nice to split not only battery, but too the visor unit itself from the CPU, to have even more flexibility.
> 
> The investment is high and we would have to start with a module like: TurboX SDW4100 Development Kit (that is alone 1000$).

Do you think it is work this inversion? Still not sure because there is a lot to do still with software.

**Comment by recom3** [comment]
We have created a new discussion to speak about the new hardware, so ideas can be discussed:

[https://github.com/recom3/api-hud-goggle/discussions/11](https://github.com/recom3/api-hud-goggle/discussions/11)

**Comment by KennethBroberg** [comment]
> I think it would be cool, but also I think that it is a very very small part of the people who own this device and actually still use it, are able to change the hardware. And yes, form factor is very important, otherwise it will not fit I guess

I think you are right on that one, Romkabouter, and I heard of many ditching the SNOW2 due lack of support from Intel, so it will almost be like a new device to create and promote.. which probably can be short-cut' by this marvelous development done.. but fearing for lawsuits by "copying" Recon/Intel properiatary technology..

and sorry for bug'ing in.. 
Hi I am Kenneth, from Denmark, and owner of the SNOW2 (Oakley on v4.5) and a Zeal (non-connected, MOD) Ski-googles.
Both are still running well, hmm .. get a BT auth error if plugged imto PC while powering up the SNOW2, but mostly when Developer mode enabled :) and "ADB shell ..." will not see the SNOW2 as connected, no devices... bugger..
and.. I will be using them from saturday next weekend in Austria (week 5) - Still got my SKIAMADA app install and runnnig on the SNOW2
and I did manage to install the gopro remote (from from another github user) via ADB/Android Studio back in the days like 2018. but do not have that pc with the developmenttools anymore, so the SNOW2 has been aple to be discovered by 'ADB then..

Let me know if I can be of any help to you guys.. 

Br K

**Comment by recom3** [comment]
> I think you are right on that one, Romkabouter, and I heard of many ditching the SNOW2 due lack of support from Intel, so it will almost be like a new device to create and promote.. which probably can be short-cut' by this marvelous development done.. but fearing for lawsuits by "copying" Recon/Intel proprietary technology..

There will be no copy as it would be most probably a CPU box with a pluggable display (micro display or wave guide, lets see). That is general and now proprietary of any one.

All the software would be open sourced, just to avoid some guys sell you a 600$ product and let it without support in few years...

I am interested in the GoPro remote, perhaps can you give us more info: for witch version of Go Pro works (does it works for newer versions?), is it via WiFi or Bluetooth, which is the link of the app in GitHub (so maybe we can integrate in current recom3 market place). We are going to release an App for the old Contour camera, that is still used by some people.

It would be very helpful if you promote our site [www.recom3.com](www.recom3.com) with your contacts maybe.

Thx very much!

**Comment by Romkabouter** [comment]
> I am interested in the GoPro remote, perhaps can you give us more info: for witch version of Go Pro works (does it works for newer versions?), is it via WiFi or Bluetooth, which is the link of the app in GitHub (so maybe we can integrate in current recom3 market place). We are going to release an App for the old Contour camera, that is still used by some people.

probably this one https://github.com/b1g0n3/MyCamRemote

**Comment by KennethBroberg** [comment]
Recom3, ahh.. ok.. yup, then it is a new product and noone can claim.. 

There is a Jonathan K, in Manchester who has done some own core development too, his own android breed for the SNOW2 as I understood on our conversations.. bootloading his own system.. I'll reach out to him too

Romkabouter.. spot on, that was the goproremote.. but I also got the Oakley version. .but since Gopro changed to ue their own app as proxy, none of those after GoPro H5 or 6 will work (thnk H5 was tha last on old setup..
I'm a heavy Insta360 camera user and might see if I can fix a small remote-control app for those..

If you can use a copy of the Oakley GoPro APK then let me know.. I got that copied out earlier.. and I just noticed tht I have a Garmin Virb remote control app as APK too

If I get my ADB kit to recognie the SNOW2 again, I can pull the GyroMaze APK from them too.. it a game where you control a rolling ball in a maze by tilting the goggles/head.. like the old wooden frame game most have tried as kid

/K

**Comment by recom3** [comment]
> Recom3, ahh.. ok.. yup, then it is a new product and noone can claim..
> 
> There is a Jonathan K, in Manchester who has done some own core development too, his own android breed for the SNOW2 as I understood on our conversations.. bootloading his own system.. I'll reach out to him too
> 
> Romkabouter.. spot on, that was the goproremote.. but I also got the Oakley version. .but since Gopro changed to ue their own app as proxy, none of those after GoPro H5 or 6 will work (thnk H5 was tha last on old setup.. I'm a heavy Insta360 camera user and might see if I can fix a small remote-control app for those..
> 
> If you can use a copy of the Oakley GoPro APK then let me know.. I got that copied out earlier.. and I just noticed tht I have a Garmin Virb remote control app as APK too
> 
> If I get my ADB kit to recognie the SNOW2 again, I can pull the GyroMaze APK from them too.. it a game where you control a rolling ball in a maze by tilting the goggles/head.. like the old wooden frame game most have tried as kid
> 
> /K

Yes, please any APK we can use as starting point, if you can share it.

And please try to contact you friend from Manchester and let see.

Many thx again!

**Comment by KennethBroberg** [comment]
Recom3, done... lets see if/when he replies..

https://drive.google.com/drive/folders/1NQYgcD7TKrOaP1V2QbKSwBCEjOVN4f6f?usp=sharing

and at above link the 2 apk's can be found..   one is the gopro remote from above, and the other is the Oakley Garmin VIrb remote.. both pulled out of the SNOW2 when I had access.. (file dates are not the right ones, somehow changed when I moved the files from a backup SSD drive)

Happy hunting for info

**Comment by AlexisNuez** [comment]
Hi,

I'm Alexis from Barcelona. I own an Oakley Airwave 1.5. Two years ago, I encountered a battery issue, and they reset the firmware to its original version. Additionally, they requested reactivation.

I have a few questions:

Do you have the firmware upgrade available for installation, and how can I install it?
I attempted to register on Recom3, but I believe I made a mistake.
Furthermore, I do not have any BitCoins.
Thank you for assisting me.

Alexis

**Comment by recom3** [comment]
Hello,
In first place, many thanks for your attention.
Yes, we have the firmware, though you can use them with our soft to track trips, find buddies, etc.
Main difference with the updated version, is that the second one allows titled maps, that you can produce yourself for any area of interest (version 1 only allow to download certain resorts).
Yup sorry to register you have to spend some Binance coin not Bitcoin.
That is because this is a private initiative not supported by anyone and we have to pay for servers etc. 
Even though our main costs are the soft it self that we give for free here in Github and you can compile to make it work for you.
Feel free to ask anything else.
Regards.

**Comment by recom3** [comment]
**April Update**

We have almost ready the update for Snow 2 Googles to the last version of software 4.5.
In this way, all factory reset Goggles will be updated to last known version and of course activated!
The update will be done with the Uplink Desktop.

**Comment by Romkabouter** [comment]
> **April Update**
> 
> We have almost ready the update for Snow 2 Googles to the last version of software 4.5. In this way, all factory reset Goggles will be updated to last known version and of course activated! The update will be done with the Uplink Desktop.

That would be great, I have an old one lying around here!

**Comment by recom3** [comment]
The update to snow2 version 4.5 is available in Desktop Uplink in private area from now on.
Some of the features of 4.5:
Wifi
Start day feature (just not to have GPS on all the time)
Maps v2 (can be produced by yourself using Uplink Desktop app)

**Comment by Romkabouter** [comment]
> The update to snow2 version 4.5 is available in Desktop Uplink in private area from now on.

I am gonna try it soon, although my season is already over but that leaves more time to get it done for next year. I can use the other one for a friend

**Comment by recom3** [comment]
> Hi,
> 
> I'm Alexis from Barcelona. I own an Oakley Airwave 1.5. Two years ago, I encountered a battery issue, and they reset the firmware to its original version. Additionally, they requested reactivation.
> 
> I have a few questions:
> 
> Do you have the firmware upgrade available for installation, and how can I install it? I attempted to register on Recom3, but I believe I made a mistake. Furthermore, I do not have any BitCoins. Thank you for assisting me.
> 
> Alexis

Hi @AlexisNuez upgrade to last version 4.5 available now. If you provide you mail (don't put the '@', just space), we can sent a pp link. Thx

**Comment by recom3** [comment]
> Recom3, done... lets see if/when he replies..
> 
> https://drive.google.com/drive/folders/1NQYgcD7TKrOaP1V2QbKSwBCEjOVN4f6f?usp=sharing
> 
> and at above link the 2 apk's can be found.. one is the gopro remote from above, and the other is the Oakley Garmin VIrb remote.. both pulled out of the SNOW2 when I had access.. (file dates are not the right ones, somehow changed when I moved the files from a backup SSD drive)
> 
> Happy hunting for info

@KennethBroberg your APK for GoPro installed and working in upgraded goggles with version snow 4.5
We will test soon with GoPro 5 and be back
Thx very much for your support

**Comment by recom3** [comment]
**May Update**

We were able to connect the **GoPro** to the goggles with upgraded 4.5 software!
A modified version of the APK provided by Kenneth was used, since it showed to work only till GoPro 4 and we used a **GoPro 5**.
Now it is tested and worked perfectly till GoPro5 and possibly 6, since it is very similar to GoPro 4 and 5.
This is beyond original implementation, that worked only with GoPro 4! 
APK  in market place:
![image](https://github.com/recom3/api-hud-goggle/assets/102611090/0a54f0bb-b9cd-437b-9865-eee589df0702)

**Comment by recom3** [comment]
**Summer update!**

Till next winter session start, a reduced life time support of 0.2 BNB will be in place.

We have also improved track display in web with an elevation profile!

Plus previous updates like Upgrade to Snow 4.5 and now also WhatsApp notifications.

**Out numbers:**

More than 800 days of recom3 site up and running.

200k lines of code in managed repos!

![image](https://github.com/recom3/api-hud-goggle/assets/102611090/99a369ab-0742-4e15-91da-155ffa2491ff)

**Comment by Romkabouter** [comment]
> > The update to snow2 version 4.5 is available in Desktop Uplink in private area from now on.
> 
> I am gonna try it soon, although my season is already over but that leaves more time to get it done for next year. I can use the other one for a friend

Hi, I have another old goggle still on version 3. I cannot find the update to version 4.5 when connected to the Desktop Uplink. Can you help me?

**Comment by recom3** [comment]
> > > The update to snow2 version 4.5 is available in Desktop Uplink in private area from now on.
> > 
> > 
> > I am gonna try it soon, although my season is already over but that leaves more time to get it done for next year. I can use the other one for a friend
> 
> Hi, I have another old goggle still on version 3. I cannot find the update to version 4.5 when connected to the Desktop Uplink. Can you help me?

Hello did not seen this thread for long time. Do you have issues yet.
Updates are working fine now.

**Comment by recom3** [comment]
Good day,

We are pleased to announce that old music app is now working in upgraded devices to 4.5 + ice3 app

![image](https://github.com/user-attachments/assets/07cc8a4d-91a7-44ef-b2ea-6252413e3167)

**Comment by AlexisNuez** [comment]
Hi, 
I would like to purchase for the all life.
Is posible to pay with another option that is not a Bitcoin.

Thank you.

**Comment by AlexisNuez** [comment]
Also have any person who live in USA.
I woul like to buy a replacement of glass fot the airwave Oakley, but is only for te US market.
If some body can buy and send me to Spain.

Thank You.
Alexis

**Comment by AlexisNuez** [comment]
Hi, I tried to pay with MetaMask. But is not easy for me.I see that I can pay by PayPal.Is possible?Can you help me.Best Regards.Alexis NuezEl 19 mar 2024, a las 22:58, recom3 ***@***.***> escribió:﻿
Hello,
In first place, many thanks for your attention.
Yes, we have the firmware, though you can use them with our soft to track trips, find buddies, etc.
Main difference with the updated version, is that the second one allows titled maps, that you can produce yourself for any area of interest (version 1 only allow to download certain resorts).
Yup sorry to register you have to spend some Binance coin not Bitcoin.
That is because this is a private initiative not supported by anyone and we have to pay for servers etc.
Even though our main costs are the soft it self that we give for free here in Github and you can compile to make it work for you.
Feel free to ask anything else.
Regards.
